akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -387,7 +387,7 @@ object Source {
    */
   def actorRef[T](bufferSize: Int, overflowStrategy: OverflowStrategy): Source[T, ActorRef] = {
     require(bufferSize >= 0, "bufferSize must be greater than or equal to 0")
-    require(overflowStrategy != OverflowStrategy.Backpressure, "Backpressure overflowStrategy not supported")
+    require(overflowStrategy != OverflowStrategies.Backpressure, "Backpressure overflowStrategy not supported")
```

Comment:
now that we have different strategy classifications we can easily push this runtime check to a type check instaed